examples/mp3.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"os/signal"
+
+	"github.com/bobertlo/go-mpg123/mpg123"
+	"github.com/gordonklaus/portaudio"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Println("missing required argument:  input file name")
+		return
+	}
+	fmt.Println("Playing.  Press Ctrl-C to stop.")
+
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, os.Kill)
+
+	// create mpg123 decoder instance
+	decoder, err := mpg123.NewDecoder("")
+	chk(err)
+
+	fileName := os.Args[1]
+	chk(decoder.Open(fileName))
+	defer decoder.Close()
+
+	// get audio format information
+	rate, channels, encoding := decoder.GetFormat()
+
+	// make sure output format does not change
+	decoder.FormatNone()
+	decoder.Format(rate, channels, encoding)
+
+	portaudio.Initialize()
+	defer portaudio.Terminate()
+	out := make([]int32, 8192)
```

Comment:
Here you effectively assume that `encoding` is `mpg123.ENC_SIGNED_32`.  You should either assert that, make it so, or handle different encodings.